src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/config/AuthAwareTokenConverter.kt

```diff
@@ -1,22 +1,22 @@
 package uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.config
 
 import org.springframework.core.convert.converter.Converter
-import org.springframework.security.authentication.AbstractAuthenticationToken
 import org.springframework.security.core.GrantedAuthority
 import org.springframework.security.core.authority.SimpleGrantedAuthority
 import org.springframework.security.oauth2.jwt.Jwt
 import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken
 import org.springframework.security.oauth2.server.resource.authentication.JwtGrantedAuthoritiesConverter
+import reactor.core.publisher.Mono
 
-class AuthAwareTokenConverter : Converter<Jwt, AbstractAuthenticationToken> {
+class AuthAwareTokenConverter : Converter<Jwt, Mono<AuthAwareAuthenticationToken>> {
   private val jwtGrantedAuthoritiesConverter: Converter<Jwt, Collection<GrantedAuthority>> =
     JwtGrantedAuthoritiesConverter()
 
-  override fun convert(jwt: Jwt): AbstractAuthenticationToken {
+  override fun convert(jwt: Jwt): Mono<AuthAwareAuthenticationToken> {
```

Comment:
why did you need to do this, since I guess it wasn't blocking before?

Reply:
not sure what you mean - the contract on the class is `Converter<Jwt, Mono<AuthAwareAuthenticationToken>>` so have to return a `Mono`

Reply:
But why is  contract `Converter<Jwt, Mono<AuthAwareAuthenticationToken>>`

and not 

`Converter<Jwt, AuthAwareAuthenticationToken>` 

?